### ISSUE TITLE
Error: failed to download "stable/nfs-server-provisioner"

### ISSUE THREAD

**Issue by stefanthorpe** [helm/helm#4080]
Trying to run a simple install but suddenly getting this message.

``helm install stable/nfs-server-provisioner```

I've previously installed other packages.

**Comment by michelleN** [comment]
hey @stefanthorpe! The nfs-server-provisioner chart is being downloaded from the `stable` repo. The first thing you'll want to do is make sure that the `helm repo list` has the following line:
```console
$ helm repo list
NAME  	URL
stable	https://kubernetes-charts.storage.googleapis.com
local 	http://127.0.0.1:8879/charts
```

If you do not see the `stable` repo listed, you can add it via `$helm repo add stable https://kubernetes-charts.storage.googleapis.com`

Can you try that out and let me know what you see?

**Comment by stefanthorpe** [comment]
That's the odd thing I have it and have used several other stable charts:
```console
stefan@stefan-XPS-13-9365:~$ helm repo list
NAME     	URL                                                      
stable   	https://kubernetes-charts.storage.googleapis.com/        
local    	http://127.0.0.1:8879/charts                             
incubator	http://storage.googleapis.com/kubernetes-charts-incubator
stefan@stefan-XPS-13-9365:~$ helm install stable/nfs-server-provisioner
Error: failed to download "stable/nfs-server-provisioner"
stefan@stefan-XPS-13-9365:~$ 
```

**Comment by carmat88** [comment]
Hello Folks! I am taking the liberty to join the thread for I was about to open a new issue, but found this one first.

We have been experiencing a similar glitch, with the command `helm upgrade` with `--install` flag though. So the attempt was the following:

`helm upgrade --install --namespace logmon --values ~/.kubernetes-yaml/logmon/prometheus-helm-values.yaml --version "v2.2.1" "prometheus" stable/prometheus`

But getting the error message for older helm version:

> Error: file "stable/prometheus" not found

Now in 2.91 updated to:

> Error: failed to download "stable/prometheus" (hint: running `helm repo update` may help)

When I test both `helm repo list` and `helm search prometheus`, here are the outputs:

```
ubuntu@ip-10-0-90-213:~$ helm repo list
NAME            	URL                                             
stable          	https://kubernetes-charts.storage.googleapis.com
local           	http://127.0.0.1:8879/charts     
```

and

```
ubuntu@ip-10-0-90-213:~$ helm search prometheus
NAME                                 	CHART VERSION	APP VERSION	DESCRIPTION                                       
stable/prometheus                    	6.7.2        	2.2.1      	Prometheus is a monitoring system and time seri...
stable/prometheus-cloudwatch-exporter	0.1.2        	0.1.0      	A Helm chart for prometheus cloudwatch-exporter   
stable/prometheus-mysql-exporter     	0.1.0        	v0.10.0    	A Helm chart for prometheus mysql exporter with...
stable/prometheus-node-exporter      	0.2.0        	0.15.2     	A Helm chart for prometheus node-exporter         
stable/prometheus-postgres-exporter  	0.2.0        	0.4.6      	A Helm chart for prometheus postgres-exporter     
stable/prometheus-pushgateway        	0.1.2        	0.4.0      	A Helm chart for prometheus pushgateway           
stable/prometheus-rabbitmq-exporter  	0.1.3        	v0.28.0    	Rabbitmq metrics exporter for prometheus          
stable/prometheus-redis-exporter     	0.1.2        	0.16.0     	Prometheus exporter for Redis metrics             
stable/prometheus-to-sd              	0.1.0        	0.2.2      	Scrape metrics stored in prometheus format and ...
stable/elasticsearch-exporter        	0.1.4        	1.0.2      	Elasticsearch stats exporter for Prometheus       
stable/stackdriver-exporter          	0.0.4        	0.5.1      	Stackdriver exporter for Prometheus               
stable/weave-cloud                   	0.2.2        	0.2.0      	Weave Cloud is a add-on to Kubernetes which pro...
stable/kube-state-metrics            	0.7.1        	1.2.0      	Install kube-state-metrics to generate and expo...
stable/mariadb                       	4.2.2        	10.1.33    	Fast, reliable, scalable, and easy to use open-...
```
The head scratching session lasted for a while! Then together with another colleague we've noticed a discrepancy between the version we were passing as an argument and the actually correct one listed above (i.e 6.7.2 instead of 2.2.1 **which is the app version, not the chart version**) By changing the passed version to the correct one did help and now we can correctly install such chart. 

Nevertheless we would like to propose to add as a return message not only the comment to *try update the repo list*, but to **also double check that a valid existing chart version number is used**.

Hope it could help and please let me know if this was actually obvious. IMHO, there is a difference between saying:

- *Error: file "stable/prometheus" not found* or *Error: failed to download "stable/prometheus"* (which is already somewhat more accurate)

and something like:

- *Error: failed to download "stable/prometheus - We couldn't found the specified chart version*

or

- *Error: failed to download "stable/prometheus - Argument X does not match current available chart's  attributes*

In other words, a sort of case in-esac structure that will return a better output message.  

Thanks and have a good one!

**Comment by michelleN** [comment]
@stefanthorpe - wow that's strange. What version of helm are you running? I'll try to reproduce. Did you also try a `helm repo update`?

@carmat88 -- thanks for your input. Agreed and it's always good to have more detailed error messages! If you'd like to contribute and them more user friendly, please feel free. 

I wonder if changing error messages like that breaks backwards compatibility. cc/ @technosophos

**Comment by aymehri** [comment]
Hello,

I was facing the same issue and was about to update this but i resolved it by adding HTTP_PROXY and HTTPS_PROXY to my command line.

Docker need to be configured with proxy. And helm need to ping https://kubernetes-charts.storage.googleapis.com from HTTP_PROXY too.

It works fine too.

**Comment by michelleN** [comment]
@stefanthorpe any updates on this? are you still seeing/experiencing this issue?

**Comment by loomsen** [comment]
Yes, I ran into this today. Calling `helm upgrade` from ansible triggered by a gitlab-ci job, it read exactly failed to download. Using `helm install` works.

**Comment by stantonxu** [comment]
I have the same issue.

```
Jiefengs-MBP:helm stanton$ helm install jenkins
Error: failed to download "jenkins" (hint: running `helm repo update` may help)
Jiefengs-MBP:helm stanton$ helm repo list
NAME  	URL
stable	https://kubernetes-charts.storage.googleapis.com
local 	http://127.0.0.1:8879/charts
```

`helm repo update` doesn't help. Reinstalling `tiller` couple times doesn't help either.

**Comment by stantonxu** [comment]
> Hello,
> 
> I was facing the same issue and was about to update this but i resolved it by adding HTTP_PROXY and HTTPS_PROXY to my command line.
> 
> Docker need to be configured with proxy. And helm need to ping https://kubernetes-charts.storage.googleapis.com from HTTP_PROXY too.
> 
> It works fine too.

@aymehri , could you provide more details? How to add HTTP_PROXY / HTTPS_PROXY to command line?

**Comment by stantonxu** [comment]
> > Jiefengs-MBP:helm stanton$ helm install jenkins
> > Error: failed to download "jenkins" (hint: running `helm repo update` may help)
> 
> @stantonxu you need to run `helm install stable/jenkins` in order to fetch the jenkins chart from the stable repository.

Thank you @bacongobbler . What a silly mistake I made.

**Comment by dkatsanikakis** [comment]
Hi everyone

I faced the same issue on MacOS X.

Adding `--debug` helped me trace the problem which as file permissions/missing folder:
`Error: open /Users/user/.helm/cache/archive/prometheus-8.8.0.tgz: permission denied`

I created the necessary folders and then it work. I also executed with sudo which also did the trick. Maybe this is not the case however i am justing adding here in case someone else hits the same issue like mine.

**Comment by bacongobbler** [comment]
Closing due to inactivity. Please open a new ticket against Helm 3 if this issue persists. Thanks!

**Comment by Nageshrathod7** [comment]
This worked for me {"helm repo add stable https://charts.helm.sh/stable"}
